frontend/www/js/omegaup/time.test.ts

```diff
@@ -1,5 +1,41 @@
 import * as time from './time';
 
+describe('getDifferenceInCalendarYears', () => {
+  // Setting an specific datetime to avoid flakiness in a leap-year
+  const now = new Date(0).getDate();
+  let dateNowSpy: jest.SpyInstance<number, []> | null = null;
+
+  beforeEach(() => {
+    dateNowSpy = jest.spyOn(Date, 'now').mockImplementation(() => now);
+  });
+
+  afterEach(() => {
+    if (dateNowSpy) {
+      dateNowSpy.mockRestore();
+    }
+  });
+
+  it('Should handle difference in calendar years', () => {
+    // https://www.epochconverter.com/
+    const yearInSeconds = 31556926 * 1000;
+    expect(
+      time.getDifferenceInCalendarYears(new Date(yearInSeconds * 1)),
+    ).toEqual(1);
+    expect(
+      time.getDifferenceInCalendarYears(new Date(yearInSeconds * 3)),
+    ).toEqual(3);
```

Comment:
```suggestion
    expect(
      time.getDifferenceInCalendarYears(new Date('2000-12-31T23:59:59Z')),
    ).toEqual(0);
    expect(
      time.getDifferenceInCalendarYears(new Date('2001-01-01T00:00:00Z')),
    ).toEqual(1);
```

here it's better to only use human-readable dates. same below.